test/distributed/org/apache/cassandra/distributed/test/SSTableLoaderEncryptionOptionsTest.java

```diff
@@ -112,7 +113,7 @@ public void bulkLoaderCannotAgreeOnClientTLSProtocol()
                                                             "--truststore-password", validTrustStorePassword,
                                                             "test/data/legacy-sstables/na/legacy_tables/legacy_na_clust");
         assertNotEquals(0, tool.getExitCode());
-        assertTrue(tool.getStdout().contains("SSLHandshakeException"));
+        assertTrue(tool.getStdout().contains("com.datastax.driver.core.exceptions.TransportException"));
```

Comment:
This doesn't make sense, the driver doesn't connect to the storage port.

Reply:
@driftx this code belongs to the test method `bulkLoaderCannotAgreeOnClientTLSProtocol` which specified `--ssl-protocol` to be `TLSv1` which is invalid as in config it is `TLSv1.2` for _client_ encryption. Bulkloader internally connects via CQL to a node, fetching metadata, getting token ranges so on before actually loading sstables. So this exception comes from the fact that we have not been able to talk to that node via CQL because we have set invalid `--ssl-protocol` flag. Basically, if fails way sooner it gets to sstable loading.

Reply:
What is puzzling is that when I checkout cassandra-4.0 as is now, without this patch applied, running this test in IDEA, locally, fails as well, huh. It is actually good question why do I need to suddenly change this assert to check if the log contains some other exception. Should be the same as it was before. But this test fails even as-is in cassandra-4.0.

Would you do me a favor a run this test yourself locally on cassandra-4.0 to see if it passes there for you? Thanks a lot.

Reply:
This fails for me too, both in isolation and in the suite.  It's a bit worrisome this doesn't appear in jenkins or circle

Reply:
Thanks, interesting, I will try to poke around whats up.

Reply:
I think that catching whatever exception we get from initialisation and wrapping it into ConfigurationException and throwing that one would do the job. At least we would be sure that it throws one exception type no matter what is the cause.